File: cpp/z2.cpp
#include <iostream>
#include <vector>

using namespace std;


bool recurs(int& check, vector<int> nomer, const int& k, string&result) {
	if (nomer.size() == 0) check++;
	if (check == k) return true;
	for (size_t i = 0; i < nomer.size(); i++) {
		vector<int> oleg = {};
		bool deleted = false;
		for (size_t j = 0; j < nomer.size(); j++) {
			if (nomer[j] != nomer[i] || deleted) {
				oleg.push_back(nomer[j]);
			}
			else deleted = true;
		}
		if (recurs(check, oleg, k, result)) {
			char k = nomer[i] + '0';
			result.push_back(k);
			return true;
		}
	}
	return false;
}

int main() {
	int n;
	int k;
	int check = 0;

	string result;

	cin >> n;
	cin >> k;

	vector<int> nomer;
	for (int i = 1; i <= n; i++) {
		nomer.push_back(i);
	}

	recurs(check, nomer, k, result);
	for (int i = n - 1; i >= 0; i--) cout << result[i];
}

File: python/z2.py
def recurs(check, nomer, k, result):
    if len(nomer) == 0:
        check[0] += 1
    if check[0] == k:
        return True
    for i in range(len(nomer)):
        oleg = []
        deleted = False
        for j in range(len(nomer)):
            if nomer[j] != nomer[i] or deleted:
                oleg.append(nomer[j])
            else:
                deleted = True
        if recurs(check, oleg, k, result):
            result.append(str(nomer[i]))
            return True
    return False

def main():
    n = int(input())
    k = int(input())
    check = [0]
    result = []

    nomer = list(range(1, n+1))

    recurs(check, nomer, k, result)
    for i in range(n - 1, -1, -1):
        print(result[i], end="")

if __name__ == "__main__":
    main()
